### PR TITLE
drivechain_client: address book

### PR DESCRIPTION
**Pull request by arnemolland**
This change introduces an address book feature currently implemented with `shared_preferences` and stored locally.

- [x] Address book CRUD
- [x] Address book modal
- [ ] Create address entry from UI
- [ ] Select/create addresses from send page

### CODE REVIEW

File: packages/drivechain_client/lib/widgets/address_menu.dart [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return ViewModelBuilder<AddressMenuViewModel>.reactive(
+      viewModelBuilder: () => AddressMenuViewModel(),
+      builder: (context, model, child) {
+        return Column(
+          children: [
+            SailText.primary12(
+              'These are your Bitcoing addresses for sending payments. Always check the amount and the receiving address before sending coins.',
```
Comment (octobocto) [inline packages/drivechain_client/lib/widgets/address_menu.dart:19]: typo in bitcoing

Comment (octobocto) [reply packages/drivechain_client/lib/widgets/address_menu.dart:19]: never heard of bitcoing, what's a bitcoing?

Comment (arnemolland) [reply packages/drivechain_client/lib/widgets/address_menu.dart:19]: 1 bitcoing is the equivalent of 100,000,000 satoshing

**Comment by arnemolland** [comment]
Closing as address book functionality is not part of what we want to do

File: packages/drivechain_client/lib/widgets/address_menu.dart (hunk 1 above)
Comment (octobocto) [reply packages/drivechain_client/lib/widgets/address_menu.dart:19]: have never heard of this before

Comment (octobocto) [reply packages/drivechain_client/lib/widgets/address_menu.dart:19]: maybe for denominating amounts, but i'm pretty sure it's not used to refer to addresses

**Pull request closed without merging**